### PR TITLE
[camera]refactor more types into separate files, and add unit tests

### PR DESCRIPTION
**Pull request by hellohuanlin**
Breaking up CameraPlugin.m is part of optional cleanups discussed [in the proposal](https://docs.google.com/document/d/1Znjp-BduXIb1Ix3JOE1YKyJi1Z0d6rB-2q6Lt7j8xSo/edit#). The monolithic CameraPlugin.m makes it hard to unit test other sub-tasks, so we want to do this clean up sooner.

Refactor the following types And added unit tests for them: 
- ResolutionPreset
- VideoFormat
- FocusMode
- ExposureMode
- DeviceOrientation

For some reason the left/right orientation is flipped, but this seems to be intentional for some unknown reason: https://github.com/flutter/plugins/pull/3390/files#r793213256

No version change: just moving code

Issue here: flutter/flutter#96429

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/camera/camera/example/ios/RunnerTests/DeviceOrientationTests.m [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+@implementation DeviceOrientationTests
+
+- (void)testFLTGetUIDeviceOrientationForString {
+  XCTAssertEqual(UIDeviceOrientationPortraitUpsideDown,
+                 FLTGetUIDeviceOrientationForString(@"portraitDown"));
+  XCTAssertEqual(UIDeviceOrientationLandscapeRight,
+                 FLTGetUIDeviceOrientationForString(@"landscapeLeft"));
+  XCTAssertEqual(UIDeviceOrientationLandscapeLeft,
+                 FLTGetUIDeviceOrientationForString(@"landscapeRight"));
```
Comment (hellohuanlin) [inline packages/camera/camera/example/ios/RunnerTests/DeviceOrientationTests.m:21]: For some reason the left/right orientation is flipped, but this seems to be intentional for some unknown reason: https://github.com/flutter/plugins/pull/3390/files#r793213256

File: packages/camera/camera/example/ios/RunnerTests/DeviceOrientationTests.m [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera.Test;
+@import AVFoundation;
+#import <XCTest/XCTest.h>
```
Comment (jmagman) [inline packages/camera/camera/example/ios/RunnerTests/DeviceOrientationTests.m:7]: Prefer module imports.
```suggestion
@import XCTest;
```

File: packages/camera/camera/example/ios/RunnerTests/ExposureModeTests.m [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera.Test;
+@import AVFoundation;
+#import <XCTest/XCTest.h>
```
Comment (jmagman) [inline packages/camera/camera/example/ios/RunnerTests/ExposureModeTests.m:7]:
```suggestion
@import XCTest;
```

File: packages/camera/camera/example/ios/RunnerTests/FocusModeTests.m [hunk 4]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera.Test;
+@import AVFoundation;
+#import <XCTest/XCTest.h>
```
Comment (jmagman) [inline packages/camera/camera/example/ios/RunnerTests/FocusModeTests.m:7]:
```suggestion
@import XCTest;
```

File: packages/camera/camera/ios/Classes/DeviceOrientation.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```
Comment (jmagman) [inline packages/camera/camera/ios/Classes/DeviceOrientation.h:1]: I like the idea of splitting out things from `CameraPlugin` but this seems excessive, each header only has like 2 functions/typedef in them, and the function implementations are super simple.  There's more boilerplate than logic in some of them.  Same with the tests, I don't think we need 5 new test files each containing ~10 lines of asserts.

There's no fast rules here, it's preference, but lots of little interrelated files are hard to navigate, too.

What about a `CameraProperties.h/m` with everything you split out?

File: packages/camera/camera/ios/Classes/ExposureMode.h [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * Represents camera's exposure mode. Mirrors ExposureMode in camera.dart.
+ */
+typedef NS_ENUM(NSInteger, FLTExposureMode) {
+  FLTExposureModeAuto,
+  FLTExposureModeLocked,
+
```
Comment (jmagman) [inline packages/camera/camera/ios/Classes/ExposureMode.h:15]:
```suggestion
```

File: packages/camera/camera/ios/Classes/ExposureMode.m [hunk 7]
```diff
@@ -0,0 +1,37 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "ExposureMode.h"
+
+NSString *FLTGetStringForFLTExposureMode(FLTExposureMode mode) {
```
Comment (jmagman) [inline packages/camera/camera/ios/Classes/ExposureMode.m:7]: Missing Swift enums yet? 😉

File: packages/camera/camera/ios/Classes/DeviceOrientation.h (hunk 5 above)
Comment (hellohuanlin) [reply packages/camera/camera/ios/Classes/DeviceOrientation.h:1]: I like the `CameraProperties.h/m` approach. Will update.

File: packages/camera/camera/ios/Classes/ExposureMode.m (hunk 7 above)
Comment (hellohuanlin) [reply packages/camera/camera/ios/Classes/ExposureMode.m:7]: Nope, it's in the header file

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**